sentry/src/routes/channel.rs

```diff
@@ -392,6 +398,112 @@ pub async fn get_accounting_for_channel<C: Locked + 'static>(
     Ok(success_response(serde_json::to_string(&res)?))
 }
 
+/// `GET /v5/channel/0xXXX.../pay` request
+///
+/// Body: [`ChannelPayRequest`]
+///
+/// Response: [`SuccessResponse`]
+pub async fn channel_payout<C: Locked + 'static>(
+    req: Request<Body>,
+    app: &Application<C>,
+) -> Result<Response<Body>, ResponseError> {
+    let channel_context = req
+        .extensions()
+        .get::<ChainOf<Channel>>()
+        .expect("Request should have Channel & Chain/TokenInfo")
+        .to_owned();
+
+    let auth = req
+        .extensions()
+        .get::<Auth>()
+        .ok_or(ResponseError::Unauthorized)?
+        .to_owned();
+
+    let spender = auth.uid.to_address();
+
+    let body = hyper::body::to_bytes(req.into_body()).await?;
+    let to_pay = serde_json::from_slice::<ChannelPayRequest>(&body)
+        .map_err(|e| ResponseError::FailedValidation(e.to_string()))?;
+
+    let channel_campaigns = fetch_campaign_ids_for_channel(
+        &app.pool,
+        channel_context.context.id(),
+        app.config.campaigns_find_limit,
+    )
+    .await?;
+
+    let campaigns_remaining_sum = app
+        .campaign_remaining
+        .get_multiple(&channel_campaigns)
+        .await?
+        .iter()
+        .sum::<Option<UnifiedNum>>()
+        .ok_or_else(|| {
+            ResponseError::BadRequest("Couldn't sum remaining amount for all campaigns".to_string())
+        })?;
+
+    // A campaign is closed when its remaining == 0
+    // therefore for all campaigns for a channel to be closed their total remaining sum should be 0
+    if campaigns_remaining_sum > UnifiedNum::from_u64(0) {
+        return Err(ResponseError::FailedValidation(
+            "All campaigns should be closed or have no budget left".to_string(),
+        ));
+    }
+
+    let accounting_spent = get_accounting(
+        app.pool.clone(),
+        channel_context.context.id(),
+        spender,
+        Side::Spender,
+    )
+    .await?
+    .map(|accounting_spent| accounting_spent.amount)
+    .unwrap_or_default();
+    let accounting_earned = get_accounting(
+        app.pool.clone(),
+        channel_context.context.id(),
+        spender,
+        Side::Earner,
+    )
+    .await?
+    .map(|accounting_spent| accounting_spent.amount)
+    .unwrap_or_default();
+    let latest_spendable =
+        fetch_spendable(app.pool.clone(), &spender, &channel_context.context.id())
+            .await
+            .map_err(|err| ResponseError::BadRequest(err.to_string()))?
+            .ok_or(ResponseError::BadRequest("Spendable not found".to_string()))?;
+    let total_deposited = latest_spendable.deposit.total;
+
+    let available_for_return = total_deposited
+        .checked_add(&accounting_earned)
+        .ok_or_else(|| ResponseError::FailedValidation("No more budget remaining".to_string()))?
```

Comment:
should better say that it has overflown when calculating the available amount and similarly - underflow for the subtraction.